render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb24ImageRepresentationHandler.java

```diff
@@ -18,23 +18,47 @@
 import java.awt.image.DataBuffer;
 import java.io.IOException;
 import java.util.Map;
+
 import javax.imageio.stream.ImageInputStream;
+
 import org.codice.imaging.nitf.render.ImageMask;
 
 class Rgb24ImageRepresentationHandler implements ImageRepresentationHandler {
     private final Map<Integer, Integer> bandMapping;
 
+    private final int numOfReadsPerBand;
+
+    private final int bitsToDiscard;
+
     private static final int ALPHA_MASK = 0xFF000000;
 
-    Rgb24ImageRepresentationHandler(final Map<Integer, Integer> bandMap) {
+    Rgb24ImageRepresentationHandler(final Map<Integer, Integer> bandMap,
+            final int actualBitsPerPixelPerBand) {
         this.bandMapping = bandMap;
+        this.numOfReadsPerBand = (int) Math.ceil(actualBitsPerPixelPerBand / ((double) Byte.SIZE));
```

Comment:
This may not be the right logic for NBPP==12. Does your test data cover that case, or only 8 and 16?